### PR TITLE
Expose GetAccountInfo to the user

### PR DESCRIPTION
**Pull request by adreed-msft**
This is required for azcopy to handle SetTier errors more gracefully.

### CODE REVIEW

File: azblob/url_service.go [hunk 1]
```diff
@@ -38,6 +38,11 @@ func (s ServiceURL) GetUserDelegationCredential(ctx context.Context, info KeyInf
 	return NewUserDelegationCredential(strings.Split(s.client.url.Host, ".")[0], *udk), nil
 }
 
+func (s ServiceURL) GetAccountInfo(ctx context.Context) (*ServiceGetAccountInfoResponse, error) {
```
Comment (zezha-msft) [inline azblob/url_service.go]: >GetAccountInfo [](start = 20, length = 14)

this API works on container and blob too

File: azblob/zt_url_service_test.go [hunk 2]
```diff
@@ -9,6 +9,14 @@ import (
 	chk "gopkg.in/check.v1" // go get gopkg.in/check.v1
 )
 
+func (s *aztestsSuite) TestGetAccountInfo(c *chk.C) {
+	sa := getBSU()
+
+	// Ensure the call succeeded. Don't test for specific account properties because we can't/don't want to set account properties.
```
Comment (zezha-msft) [inline azblob/zt_url_service_test.go]: >// Ensure the call succeeded. Don't test for specific account properties because we can't/don't want to set account properties. [](start = 1, length = 127)

perhaps validate that it's at least not empty

**Comment by zezha-msft** [comment]
https://docs.microsoft.com/en-us/rest/api/storageservices/get-account-information#request

File: azblob/url_service.go (hunk 1 above)
Comment (zezha-msft) [reply azblob/url_service.go]: https://docs.microsoft.com/en-us/rest/api/storageservices/get-account-information#request

File: go.mod [hunk 3]
```diff
@@ -5,3 +5,5 @@ require (
 	github.com/kr/pretty v0.1.0 // indirect
 	gopkg.in/check.v1 v1.0.0-20180628173108-788fd7840127
 )
+
```
Comment (JohnRusk) [inline go.mod]: Minor: given that this is a public SDK, it would be better if any Go version change was in a separate PR, instead of here.

**Comment by adreed-msft** [comment]
Squashing and merging.
